backend/db/__init__.py

```diff
@@ -30,4 +35,213 @@ def getSecretary(self, email: str) -> Union[None, dict]:
\ No newline at end of file
+			return result
+
+
+	def getAllCountries(self) -> dict:
+
+		sql = "select code, name from country where is_active = true"
+		result = {"countries":[]}
+		try:
+			with self._getConnection() as dbConn:
+				with dbConn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+					cursor.execute(sql)
+					tmp = cursor.fetchone()
+					while tmp:
+						result["countries"].append({"name":tmp[1], "code":tmp[0]})
+						tmp = cursor.fetchone()
+			self._releaseConnection(dbConn)
+		except psycopg2.DatabaseError as error:
+			# TODO: logging
+			# TODO: define standard for database error messages
+			print(error)
+		finally:
+			# print(result)
+			return result
+
+
+	def getAllSports(self) -> dict:
+
+		sql = "select code, title from sport"
+		result = {"sports":[]}
+		try:
+			with self._getConnection() as dbConn:
+				with dbConn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+					cursor.execute(sql)
+					tmp = cursor.fetchone()
+					while tmp:
+						result["sports"].append({"title":tmp[1], "code":tmp[0]})
+						tmp = cursor.fetchone()
+			self._releaseConnection(dbConn)
+		except psycopg2.DatabaseError as error:
+			# TODO: logging
+			# TODO: define standard for database error messages
+			print(error)
+		finally:
+			# print(result)
+			return result
+
+	def getInactiveCountries(self) -> dict:
+
+		sql = "select code, name from country where is_active = false"
+		result = {"countries": []}
+		try:
+			with self._getConnection() as dbConn:
+				with dbConn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+					cursor.execute(sql)
+					tmp = cursor.fetchone()
+					while tmp:
+						result["countries"].append({"name": tmp[1], "code": tmp[0]})
+						tmp = cursor.fetchone()
+			self._releaseConnection(dbConn)
+		except psycopg2.DatabaseError as error:
+			# TODO: logging
+			# TODO: define standard for database error messages
+			print(error)
+		finally:
+			# print(result)
+			return result
+
+
+	def getBranchesWithSports(self) -> dict:
+
+		sql = "select s.code, s.title, b.code, b.title from sport s join branch b on b.sport_id = s.id"
+		result = {"sports": []}
+		try:
+			with self._getConnection() as dbConn:
+				with dbConn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+					cursor.execute(sql)
+					tmp = cursor.fetchone()
+					while tmp:
+						result["sports"].append({"sport_code": tmp[0], "sport_title": tmp[1], "branch_code": tmp[2], "branch_title": tmp[3]})
+						tmp = cursor.fetchone()
+			self._releaseConnection(dbConn)
+		except psycopg2.DatabaseError as error:
+			# TODO: logging
+			# TODO: define standard for database error messages
+			print(error)
+		finally:
+			#print(result)
+			return result
+
+
+	def getFundingData(self, country_code: str) -> dict:
+		sql = f"select b.title, f.absolute_funding, f.currency from funding f cross join country c join branch b on c.code = '{country_code}' and f.country_id = c.id and b.id = f.branch_id"
```

Comment:
@Sabka 
Pridávať argumenty resp. parametre do sql týmto 'priamym' spôsobom je nebezpečné (sql injuction).
Psycopg2 uvádzajú ako správny spôsob
(https://www.psycopg.org/docs/usage.html#passing-parameters-to-sql-queries),
teda
<pre>
sql = f"select b.title, f.absolute_funding, f.currency from funding f cross join country c join branch b on c.code = %s and f.country_id = c.id and b.id = f.branch_id"
...
cursor.execute(sql, (country_code,))  # <-> cursor.execute(sql, [country_code])
</pre>
Prosím zmeň to vo všetkých metódach, v ktorých sa pridáva nejakú parameter.

Reply:
jo, na to som uplne zabudla, idem to opravit

Reply:
je to done aj otestovane